niri-ipc/src/socket.rs

```diff
@@ -1,73 +1,63 @@
-use std::io::{self, Write};
+//! Helper for blocking communication over the niri socket.
+
+use std::env;
+use std::io::{self, Read, Write};
+use std::net::Shutdown;
 use std::os::unix::net::UnixStream;
 use std::path::Path;
 
-use serde_json::de::IoRead;
-use serde_json::StreamDeserializer;
-
 use crate::{Reply, Request};
 
 /// Name of the environment variable containing the niri IPC socket path.
 pub const SOCKET_PATH_ENV: &str = "NIRI_SOCKET";
 
-/// A client for the niri IPC server.
+/// Helper for blocking communication over the niri socket.
 ///
 /// This struct is used to communicate with the niri IPC server. It handles the socket connection
 /// and serialization/deserialization of messages.
-pub struct NiriSocket {
+pub struct Socket {
```

Comment:
I personally don't like this generic name as much. When `use`d in a file, it's easier to confuse with more general I/O operations when skimming (compare to naming of `UnixStream`, `TcpStream`, `UdpSocket` in the std library). It's an unimportant point though; this was an exhaustive list of all the downsides that such a name brings.

Reply:
I get that, but I decided to do this because in other niri code I'm already using fully-qualified names for some types like `niri_config::Animation` and `niri_ipc::Output` (leaving non-qualified names for more frequently occurring types of the same name). So it's consistent this way.